lib/rules/no-extra-semicolons/index.js

```diff
@@ -54,16 +54,29 @@ const rule = function (actual) {
     }
 
     root.walk(node => {
-      const rawBeforeNode = node.raws.before
+      let rawBeforeNode = node.raws.before
 
       if (rawBeforeNode && rawBeforeNode.trim().length !== 0) {
         let allowedSemi = 0
 
+        const next = node.next()
+
         // Forbid semicolon before first custom properties sets
-        if (isCustomPropertySet(node) && node.parent.index(node) > 0) {
+        // Also ignore comment before custom properties sets
+        if ((isCustomPropertySet(node) && node.parent.index(node) > 0)
+          || (node.type === "comment" && next
+            && (isCustomPropertySet(next) && node.parent.index(next) > 0 || next.type === "comment"))
+        ) {
           allowedSemi = 1
         }
 
+        const prev = node.prev()
+
+        if (isCustomPropertySet(node) && node.parent.index(node) > 0 && prev && prev.type === "comment") {
+          rawBeforeNode = prev.toString() + rawBeforeNode
```

Comment:
@hudochenkov i do this in another PR :smile: https://github.com/stylelint/stylelint/pull/2396#issuecomment-283448628